packages/frontend-shared/src/gql-components/HeaderBarContent.vue

```diff
@@ -139,12 +146,76 @@ const clearActiveProject = () => {
   }
 }
 
+const savedState = computed(() => {
+  return props.gql?.app?.activeProject?.savedState
+})
+
+const cloudProjectId = computed(() => {
+  return props.gql?.app?.activeProject?.config?.find((item: { field: string }) => item.field === 'projectId')?.value
+})
+
 const props = defineProps<{
   gql: HeaderBar_HeaderBarContentFragment,
   showBrowsers?: boolean,
-  pageName?: string
+  pageName?: string,
+  allowAutomaticPromptOpen?: boolean
 }>()
 
 const { t } = useI18n()
 
+const prompts = sortBy([
+  {
+    slug: 'ci1',
+    interval: interval('4 days'),
+    noProjectId: true,
+  },
+  {
+    slug: 'orchestration1',
+    noProjectId: true,
+  },
+], 'interval')
+
+watch(savedState, (newVal) => {
```

Comment:
Instead of creating a ref for `promptToShow` then watching and mutating it based on `props`, shouldn't we just make `promptToShow` a computed value? 

```ts
const promptToShow = computed(() => {
  // ... various logic around shouldShowPrompt, etc ...
  return prompts[props.gql?.app?.activeProject?.savedState]
})
```

Then for `clear-force-open`, instead of setting `promptToShow = ''`, we would just have a boolean such as `shouldShowPrompt` and set it to `false`. Unless I'm missing something, it seems like this would reduce the amount of computations and watchers needed.

Basically when you combine a `watch` with a reactive value that then assigns another reactive value, you are basically just re-inventing `computed`. In this case, the prompt to show is just derived data from `props`, so we should be able to do it with a single `computed`. I think `watch` is generally useful for a side effect such as making an API call, etc, it feels like a code smell to use it to just reassign to another reactive variable. 

If it helps, `computed` also takes an object of options with `get` and `set`: https://v3.vuejs.org/guide/reactivity-computed-watchers.html#computed-values

Reply:
I've cleaned this up a bit and removed the watcher. If and when the time comes to get more sophisticated with these prompts there will be some logic to revisit around this.